game-core/src/main/java/games/strategy/triplea/ui/FlagDrawMode.java

```diff
@@ -0,0 +1,63 @@
+package games.strategy.triplea.ui;
+
+import games.strategy.triplea.ui.screen.UnitsDrawer;
+import java.util.prefs.Preferences;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Class to change how unit flags are drawn on the map. Unit flags are flags drawn underneath or
+ * below units to make it easier to identify which nation they belong to. By default flag drawing is
+ * turned off, toggling once turns small flags on, toggle again to turn on large flags and again to
+ * turn off.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+class FlagDrawMode {
+
+  static void toggleNextDrawMode(final MapPanel mapPanel) {
+
+    final Preferences prefs = Preferences.userNodeForPackage(FlagDrawMode.class);
+    final UnitsDrawer.UnitFlagDrawMode setting =
+        Enum.valueOf(
+            UnitsDrawer.UnitFlagDrawMode.class,
+            prefs.get(
+                UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
+                UnitsDrawer.UnitFlagDrawMode.SMALL_FLAG.toString()));
```

Comment:
I know this is copied code, but I'm pretty sure `UnitsDrawer.UnitFlagDrawMode.valueOf("something")` would work as well.